### PR TITLE
#66 validate test result  + #75 + #76

### PR DESCRIPTION
**Pull request by alesyalis**
+ #76 When updating a project, staff is accepted only by id
+ #75Validation of feedback types (without a test) into UpsertFeedbackValidator

### CODE REVIEW

File: Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs [hunk 1]
```diff
@@ -13,6 +14,11 @@ public UpsertFeedbackValidator()
                 .MaximumLength(500);
             RuleFor(f => f.Rating)
                 .NotEmpty();
+            RuleFor(f => f.Type)
+                .IsInEnum()
+                .WithMessage("Type doesn't exist")
+                .Must(f => !new[] { FeedbackTypeDTO.Test }.Contains(f))
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs:20]: t => t != FeedbackTypeDTO.Test

File: Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs [hunk 2]
```diff
@@ -13,6 +14,11 @@ public UpsertFeedbackValidator()
...
+                .WithMessage("Cannot change to selected type manually");
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs:21]: Cannot create/update test feedback

File: Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs]: done

File: Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs (hunk 2 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Validators/UpsertFeedbackValidator.cs]: done

**Review by elizavetapavliv — APPROVED** [review]

File: Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs [hunk 3]
```diff
@@ -17,7 +17,12 @@ public ProjectProfile()
                 .ForMember(dest => dest.Mentors, conf => conf.MapFrom(src => src.Mentors.Select(m => new Staff() { UserId = m })))
                 .ForMember(dest => dest.Interviewers, conf => conf.MapFrom(src => src.Interviewers.Select(i => new Staff() { UserId = i })))
                 .ForMember(dest => dest.Recruiters, conf => conf.MapFrom(src => src.Recruiters.Select(r => new Staff() { UserId = r })));
-            CreateMap<Project, ProjectDTO>().ReverseMap();
+            CreateMap<ProjectDTO, Project>().ReverseMap();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs:20]: you don't need ProjectDTO -> Project map, only reversed

File: Recrutify/Recrutify.Services/DTOs/ProjectDTO.cs [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+using Recrutify.DataAccess.Models;
 
 namespace Recrutify.Services.DTOs
 {
     public class ProjectDTO : BaseProjectDTO
     {
         public Guid Id { get; set; }
 
-        public IEnumerable<StaffDTO> Managers { get; set; }
+        public IEnumerable<Staff> Managers { get; set; }
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/DTOs/ProjectDTO.cs:11]: why Staff and not StaffDTO, we don't use models in dtos

Comment (alesyalis) [reply Recrutify/Recrutify.Services/DTOs/ProjectDTO.cs]: done

File: Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs (hunk 3 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/Profiles/ProjectProfile.cs]: done

File: Recrutify/Recrutify.Services/DTOs/ProjectDTO.cs [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using Recrutify.DataAccess.Models;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/DTOs/ProjectDTO.cs:3]: remove

Comment (alesyalis) [reply Recrutify/Recrutify.Services/DTOs/ProjectDTO.cs]: done

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**